Doc review: Update Release 221121.7 (#1596)
SAP/spartacus-docs: _pages/home/updating-to-version-221121.7/modernize-app-to-standalone-bootstrap-application.md

---
title: Modernizing Your Storefront (Manual Fallback Steps Only)
---

The procedures on this page are provided only in case you encountered an issue while running the Spartacus schematics for modernizing your app to use the Angular `bootstrapApplication()` API and convert your root `AppComponent` to a standalone component, as described in [Modernizing Your Storefront to Use the Standalone Bootstrap Application](migration.md#modernizing-your-storefront-to-use-the-standalone-bootstrap-application). If you have already successfully run those schematics, you can ignore the steps below.

> 1. In `src/app/app.component.ts`, convert your root `AppComponent` to a storefront component by removing `standalone: false` and adding the necessary imports of Spartacus `StorefrontComponent`.

Reviewer: ```suggestion
1. In `src/app/app.component.ts`, convert your root `AppComponent` to a Standalone component by removing `standalone: false` and adding the necessary imports of Spartacus `StorefrontComponent`.
```

Reviewer 2: I'd go with `standalone` instead of `Standalone` starting with a capital letter.

Author: Oh man, I must have been getting tired 🙄 😮‍💨 
Yes, definitely standalone, not storefront, and I checked Angular docs, they are writing it with a lowercase "s", "standalone".
Thank you for catching this! 😅

   The following is an example of how your `app.component.ts` file should look:

   ```ts
   import { StorefrontComponent } from '@spartacus/storefront';
   
    @Component({
     selector: 'app-root',
     imports: [StorefrontComponent],
    })
    export class AppComponent {
   ```

   **Note:** The root component must be standalone to work with the modern Angular `bootstrapApplication()` API.

2. In `src/app/app.module.ts`, remove the following Angular configurations from your `AppModule`:

   - remove `bootstrap: [AppComponent]` from `@NgModule`
   - remove `declarations: [AppComponent]` from `@NgModule`
   - remove the following `providers` from `@NgModule`:
     - `provideBrowserGlobalErrorListeners()`
     - `provideZoneChangeDetection({ eventCoalescing: true })`
     - `provideHttpClient(withFetch(), withInterceptorsFromDi())`
   - remove `BrowserModule` from the `imports` array of `@NgModule`

   The following is an example of how your `app.module.ts` file should look:

   ```ts
    @NgModule({
     imports: [
       StoreModule.forRoot({}),
       AppRoutingModule,
       EffectsModule.forRoot([]),
       SpartacusModule
     ],
     providers: [

>      ],

Reviewer: The imports and providers array might contain other customer's custom items. that's why in the original code snippet I added `/*...*/`.

    })
    export class AppModule { }
   ```

   **Note:** In the next step, the Angular configurations are moved to a new file, named `app.config.ts`.

3. Create a new file called `src/app/app.module.ts`.
4. In the new `app.module.ts` file, which will contain the Angular configurations previously located in `AppModule`, import what remains in the `AppModule` using `importProvidersFrom(AppModule)`.

   The following is an example of how your `app.module.ts` file should look:

> 

Reviewer: typo. it should be app.config

```suggestion
3. Create a new file called `src/app/app.config.ts`.
4. In the new `app.config.ts` file, which will contain the Angular configurations previously located in `AppModule`, import what remains in the `AppModule` using `importProvidersFrom(AppModule)`.

   The following is an example of how your `app.config.ts` file should look:

```

   Specifically, remove `export { AppServerModule as default } from './app/app.module.server';`, and otherwise update `main.server.ts` to look as follows:

   ```ts
   import { BootstrapContext, bootstrapApplication } from '@angular/platform-browser';
   import { AppComponent } from './app/app.component';
   import { config } from './app/app.config.server';
   
   const bootstrap = (context: BootstrapContext) =>
     bootstrapApplication(AppComponent, config, context);
   
   export default bootstrap;
   ```

5. In `src/server.ts`, rename the default import from `AppServerModule` to the `bootstrap` function, and update the `ngExpressEngine` configuration accordingly.

   Specifically, remove `import AppServerModule from './main.server';`, remove `bootstrap: AppServerModule,` from `ngExpressEngine`, and otherwise update `server.ts` to look as follow:

   ```ts
   import bootstrap from './main.server';
   
    /* ... */
   
       ngExpressEngine({
        bootstrap,
       })
   ```

6. In `src/app/app.module.ts`, remove the configuration of non-destructive client hydration from `AppModule`.

>    Specifically, remove `provideClientHydration(withEventReplay(), withNoHttpTransferCache())` from the `providers` array. Your `app.module.ts` file should look as follows:
> 
>    ```ts
>     @NgModule({
>       /* ... */
>       providers: [
>         /* ... */
>       ],
>     })
>     export class AppModule { }
>    };
>    ```

Reviewer: ```suggestion
   Specifically, remove `provideClientHydration(withEventReplay(), withNoHttpTransferCache())` from the `providers` array. Your `app.module.ts` file should look as follows:

   ```ts
    @NgModule({
      /* ... */
      providers: [
        /* ... there should be no `provideClientHydration(withEventReplay(), withNoHttpTransferCache())` */
      ],
    })
    export class AppModule { }
   };
   ```
```

Author: Perfect! 👍 🙂

   **Note:** The configuration of non-destructive client hydration is moved to `app.config.ts` in the next step.

7. In `src/app/app.config.ts`, add the configuration of non-destructive client hydration, as shown in the following example:

   ```ts
    export const appConfig: ApplicationConfig = {
      providers: [
        /* ... */
        provideClientHydration(withEventReplay(), withNoHttpTransferCache())
     ]
   };
   ```

   You have now modernized your app to use Angular's `bootstrapApplication()` API and converted your root `AppComponent` to a standalone component, aligning it with current Angular best practices.

>    Although not required, it is also recommended that you migrate your custom components to standalone components. For more information, see [Migrating to Standalone Components in Spartacus](standalone-components-in-spartacus.md).

Reviewer: ```suggestion
   It is also strongly recommended that you migrate your custom components to standalone components. For more information, see [Migrating to Standalone Components in Spartacus](standalone-components-in-spartacus.md).
```

Author: Yes!! Good catch!! 👍 🙂
